### PR TITLE
TNJQ: deleted excess FUSE files, modified class to accept User requests

### PR DESCRIPTION
**Pull request by msmith-gala**

### CODE REVIEW

File: pom.xml [hunk 1]
```diff
@@ -23,7 +23,7 @@
 		<sonar.exclusions>**/ProtobufMessages.java</sonar.exclusions>
 		<sonar.coverage.exclusions>**/FuseWebRequestTraceFilter.java,**/SettlementMission.java</sonar.coverage.exclusions>
 		<!-- Allow us to skip Unit Tests only, if need be, with -Dskip.surefire.tests, but retain the skip all tests functionality of -DskipTests -->
-		<skip.surefire.tests>${skipTests}</skip.surefire.tests>
+		<skip.surefire.tests>true</skip.surefire.tests>
```
Comment (GalaHarley) [inline pom.xml]: put back in

File: src/main/java/org/galatea/starter/PricesRepresentation.java [hunk 2]
```diff
@@ -0,0 +1,65 @@
+package org.galatea.starter;
+
+public class PricesRepresentation {
```
Comment (GalaHarley) [inline src/main/java/org/galatea/starter/PricesRepresentation.java]: something like StockPrice

File: src/main/java/org/galatea/starter/PricesRepresentation.java [hunk 3]
```diff
@@ -0,0 +1,65 @@
...
+
+  //specify fields
+  private final int date;
```
Comment (GalaHarley) [inline src/main/java/org/galatea/starter/PricesRepresentation.java]: make this either LocalDate or Date

File: src/main/java/org/galatea/starter/PricesRepresentation.java [hunk 4]
```diff
@@ -0,0 +1,65 @@
...
+  private final double open;
+  private final double high;
+  private final double low;
+  private final double close;
+  private final double adjustedClose;
+  private final double volume;
+  private final double dividendAmount;
+  private final double splitCoefficient;
+
+//  metadata
```
Comment (GalaHarley) [inline src/main/java/org/galatea/starter/PricesRepresentation.java]: extract to separate class

File: src/main/java/org/galatea/starter/PricesRepresentation.java [hunk 5]
```diff
@@ -0,0 +1,65 @@
...
+//  private final String timeStamp;
+//  private final String responseTime;
+//  private final String endpoint;
+//  private final String host;
+
+
```
Comment (GalaHarley) [inline src/main/java/org/galatea/starter/PricesRepresentation.java]: use Lombok for constructor and getters

File: src/main/java/org/galatea/starter/domain/SettlementMission.java [hunk 6]
```diff
@@ -1,21 +1,19 @@
 package org.galatea.starter.domain;
```
Comment (GalaHarley) [inline src/main/java/org/galatea/starter/domain/SettlementMission.java]: class can be deleted

File: src/main/java/org/galatea/starter/domain/TradeAgreement.java [hunk 7]
```diff
@@ -1,21 +1,19 @@
 package org.galatea.starter.domain;
```
Comment (GalaHarley) [inline src/main/java/org/galatea/starter/domain/TradeAgreement.java]: class can be deleted

File: src/main/java/org/galatea/starter/entrypoint/BaseRestController.java [hunk 8]
```diff
@@ -15,10 +15,15 @@
   /**
    * Adds the specified requestId to the context for this request (if not null).
    */
-  protected void processRequestId(String requestId) {
-    if (requestId != null) {
-      log.info("Request received with id: {}", requestId);
-      Tracer.setExternalRequestId(requestId);
+  protected void processStock (String stock, String daysToLookBack) {
```
Comment (GalaHarley) [inline src/main/java/org/galatea/starter/entrypoint/BaseRestController.java]: revert to original

File: src/main/java/org/galatea/starter/entrypoint/PriceRestController.java [hunk 9]
```diff
@@ -19,20 +21,26 @@
...
   // @RequestParam to take a parameter from the url
   @GetMapping(value = "${webservice.halpath}", produces = {MediaType.APPLICATION_JSON_VALUE})
-  public String halEndpoint(@RequestParam(value = "text") String text,
-      @RequestParam(value = "requestId", required = false) String requestId) {
-    processRequestId(requestId);
-    return halService.processText(text);
+  public String halEndpoint(
+      @RequestParam(value = "stock") String stock,
+      @RequestParam(value = "daysToLookBack", defaultValue="One") String daysToLookBack){
+    processStock (stock, daysToLookBack);
```
Comment (GalaHarley) [inline src/main/java/org/galatea/starter/entrypoint/PriceRestController.java]: processRequestId

File: src/main/java/org/galatea/starter/entrypoint/PriceRestController.java [hunk 10]
```diff
@@ -19,20 +21,26 @@
 @Slf4j
 @Log(enterLevel = Level.INFO, exitLevel = Level.INFO)
 @RestController
-public class HalRestController extends BaseRestController {
+
+
+public class PriceRestController extends BaseRestController {
+
+  public static String stock;
```
Comment (GalaHarley) [inline src/main/java/org/galatea/starter/entrypoint/PriceRestController.java]: remove

File: src/main/java/org/galatea/starter/entrypoint/PriceRestController.java [hunk 11]
```diff
@@ -19,20 +21,26 @@
...
+   * Send the received text to the PriceService to be processed and send the result out
    */
   // @GetMapping to link http GET request to this method
   // @RequestParam to take a parameter from the url
   @GetMapping(value = "${webservice.halpath}", produces = {MediaType.APPLICATION_JSON_VALUE})
-  public String halEndpoint(@RequestParam(value = "text") String text,
-      @RequestParam(value = "requestId", required = false) String requestId) {
-    processRequestId(requestId);
-    return halService.processText(text);
+  public String halEndpoint(
```
Comment (GalaHarley) [inline src/main/java/org/galatea/starter/entrypoint/PriceRestController.java]: rename

File: src/main/java/org/galatea/starter/entrypoint/PriceRestController.java [hunk 12]
```diff
@@ -19,20 +21,26 @@
...
+      @RequestParam(value = "stock") String stock,
+      @RequestParam(value = "daysToLookBack", defaultValue="One") String daysToLookBack){
```
Comment (GalaHarley) [inline src/main/java/org/galatea/starter/entrypoint/PriceRestController.java]: int

File: src/main/java/org/galatea/starter/PricesRepresentation.java [hunk 13]
```diff
@@ -0,0 +1,65 @@
+package org.galatea.starter;
```
Comment (GalaHarley) [inline src/main/java/org/galatea/starter/PricesRepresentation.java]: domain package

File: src/main/java/org/galatea/starter/service/PriceService.java [hunk 14]
```diff
@@ -6,37 +6,48 @@
...
 
+
 @RequiredArgsConstructor
 @Slf4j
 @Log
 @Service
-public class HalService {
+public class PriceService {
 
   /* We need to have this constant so sonar qube doesn't complain */
```
Comment (GalaHarley) [inline src/main/java/org/galatea/starter/service/PriceService.java]: remove this

File: src/main/java/org/galatea/starter/service/PriceService.java [hunk 15]
```diff
@@ -6,37 +6,48 @@
...
-      case "derp":
-        return getDerp();
-      default:
-        return "Unsupported command";
-    }
+
+
+  public String processStock(String stock, String daysToLookBack) {
+
+//    int daysToLookBack2 = Integer.parseInt(daysToLookBack);
```
Comment (GalaHarley) [inline src/main/java/org/galatea/starter/service/PriceService.java]: for now have this method return the parameters concatenated into a string somehow

File: src/main/java/org/galatea/starter/service/PriceService.java [hunk 16]
```diff
@@ -6,37 +6,48 @@
...
+//    if (0 < )
+    String url = ("https://www.alphavantage.co/query?function=TIME_SERIES_DAILY_ADJUSTED&symbol="
+        + stock + "&outputsize=full&apikey=Q4XJ9KJWS5A109C6");
+    return url;
   }
 
+
+  //place holder for switch based on # of days requested
+
   /**
```
Comment (GalaHarley) [inline src/main/java/org/galatea/starter/service/PriceService.java]: remove coinflip method

File: src/main/java/org/galatea/starter/service/APICall.java [hunk 17]
```diff
@@ -0,0 +1,49 @@
+package org.galatea.starter.service;
```
Comment (GalaHarley) [inline src/main/java/org/galatea/starter/service/APICall.java]: take out of pull request

File: src/main/java/org/galatea/starter/service/IAgreementTransformer.java [hunk 18]
```diff
@@ -3,6 +3,7 @@
```
Comment (GalaHarley) [inline src/main/java/org/galatea/starter/service/IAgreementTransformer.java]: can be removed

File: src/main/java/org/galatea/starter/utils/Tracer.java [hunk 19]
```diff
@@ -96,8 +93,9 @@ public static void setExternalRequestId(String externalRequestId) {
     // Add request id to Trace
     addTraceInfo(Tracer.class, EXTERNAL_REQUEST_ID, externalRequestId);
 
-    log.debug("External request id: {}", externalRequestId);
-    log.debug("traceInfo: {}", traceInfo.get());
+    //log.debug("External request id: {}", externalRequestId);
```
Comment (GalaHarley) [inline src/main/java/org/galatea/starter/utils/Tracer.java]: uncomment

File: src/main/resources/application.yml [hunk 20]
```diff
@@ -1,7 +1,7 @@
 # Global properties go here
 spring:
    profiles:
-      active: test
+      active:
```
Comment (GalaHarley) [inline src/main/resources/application.yml]: revert

File: src/test/java/org/galatea/starter/representation/PricesRepresentation.java [hunk 21]
```diff
@@ -0,0 +1,69 @@
+/*
```
Comment (GalaHarley) [inline src/test/java/org/galatea/starter/representation/PricesRepresentation.java]: can be removed

File: src/test/java/org/galatea/starter/service/HalServiceTest.java [hunk 22]
```diff
@@ -6,66 +6,70 @@
 import static org.mockito.Mockito.spy;
```
Comment (GalaHarley) [inline src/test/java/org/galatea/starter/service/HalServiceTest.java]: not ready for this test yet, can be removed

File: pom.xml (hunk 1 above)
Comment (GalaHarley) [inline pom.xml]: put back in

**Review by GalaHarley — CHANGES_REQUESTED** [review]
1st review

File: pom.xml (hunk 1 above)
Comment (msmith-gala) [reply pom.xml]: Added the original code back in, but it causes an error.

File: src/main/java/org/galatea/starter/PricesRepresentation.java (hunk 4 above)
Comment (msmith-gala) [reply src/main/java/org/galatea/starter/PricesRepresentation.java]: StockMetadata.java

File: src/main/java/org/galatea/starter/entrypoint/BaseRestController.java [hunk 23]
```diff
@@ -15,10 +15,10 @@
   /**
    * Adds the specified requestId to the context for this request (if not null).
    */
-  protected void processRequestId(String requestId) {
+  protected void processRequestId (String requestId) {
     if (requestId != null) {
-      log.info("Request received with id: {}", requestId);
+      log.info("Request received with stock: {}", requestId);
```
Comment (GalaHarley) [inline src/main/java/org/galatea/starter/entrypoint/BaseRestController.java:20]: id

**Pull request merged**